components/standardizers/src/main/java/org/datacleaner/beans/standardize/CountryStandardizationTransformer.java

```diff
@@ -61,19 +76,49 @@ public OutputColumns getOutputColumns() {
     public String[] transform(InputRow inputRow) {
         final String value = inputRow.getValue(countryColumn);
         final Country country = Country.find(value, defaultCountry);
+
+        final String countryName;
         if (country == null) {
-            return new String[1];
+            countryName = null;
+        } else {
+            switch (outputFormat) {
+            case ISO2:
+                countryName = country.getTwoLetterISOCode();
+                break;
+            case ISO3:
+                countryName = country.getThreeLetterISOCode();
+                break;
+            case NAME:
+                countryName = country.getCountryName();
+                break;
+            default:
+                throw new IllegalStateException("Unexpected output format: " + outputFormat);
+            }
         }
-        switch (outputFormat) {
-        case ISO2:
-            return new String[] { country.getTwoLetterISOCode() };
-        case ISO3:
-            return new String[] { country.getThreeLetterISOCode() };
-        case NAME:
-            return new String[] { country.getCountryName() };
-        default:
-            throw new IllegalStateException("Unexpected output format: " + outputFormat);
+
+        final String correctedCountryName;
+        
+        if(countryName != null){
+            correctedCountryName = countryName;
+        } else {
+            correctedCountryName = LabelUtils.UNEXPECTED_LABEL;
         }
+        
+        // ConcurrentHashMap does not support null keys
+        synchronized (this) {
+            if (!countryCountMap.containsKey(countryName)) {
+                countryCountMap.put(correctedCountryName, _rowAnnotationFactory.createAnnotation());
+            }
+
+            _rowAnnotationFactory.annotate(inputRow, 1, countryCountMap.get(correctedCountryName));
```

Comment:
The annotate method is thread-safe but it is placed in the synchronized block. A slight improvement would be to separate the resolving of the RowAnnotation (in synchronized block) from the annotate call (outside).